trinity/nodes/light.py

```diff
@@ -60,7 +57,7 @@ def get_chain(self) -> LightDispatchChain:
 
         return self._chain
 
-    def get_p2p_server(self) -> BaseService:
+    def get_p2p_server(self) -> LightPeerChain:
```

Comment:
What is the rationale for specifying `LightPeerChain` here? I think it's only important that this method returns a `BaseService` (like the superclass's abstract method). The implementation choice shouldn't matter.

Reply:
The return value of this method ends up as `peer_chain` in [`LightDispatchChain`](https://github.com/cburgdorf/py-evm/blob/222e53d79da62e5f48214cb6d56178ca61ce350e/trinity/chains/light.py#L63) which is expecting a `LightPeerChain` not a `BaseService`. In other words, we can not pass a base class somewhere where a more concrete class is expected. (e.g. an API expecting `SportsCar` isn't satisfied with `Car` because `Car` may lack `go_real_fast()` API.

So, if you make this `BaseService` mypy will rightfully yell at you.

```
trinity/nodes/light.py:60: error: Argument "peer_chain" to "LightDispatchChain" has incompatible type "BaseService"; expected "LightPeerChain"
```